src/Parquet/Reader.hs

```diff
@@ -390,54 +456,80 @@ generateInstructions = loop Seq.empty
 -- https://blog.twitter.com/engineering/en_us/a/2013/dremel-made-simple-with-parquet.html
 mkInstructions ::
   forall m.
-  ( C.MonadResource m,
-    C.MonadIO m,
+  ( C.MonadIO m,
     C.MonadThrow m,
     MonadLogger m,
-    MonadReader TT.FileMetadata m
+    MonadReader TT.FileMetadata m,
+    MonadFail m
   ) =>
   (ColumnValue, [T.Text]) ->
   m (Maybe InstructionSet)
-mkInstructions = go 1
+mkInstructions (c, path) = do
+  logInfo $ "Creating instruction for column value: " <> T.pack (show c) <> " and path " <> T.intercalate "." path
+  !result <- go [] (c, path)
```

Comment:
Do we need to force the evaluation of result here? And if we do can we add a little comment as to why?

Reply:
Not really; I was suspicious of evaluation of this thunk due to a bug, but it didn't turn out to be the culprit & I just ended up forgetting it here. Fixing.